### DISCUSSION TITLE
Adding color to the background

### DISCUSSION THREAD

**Discussion by unbrother** [koenderks/aRtsy#155] [Q&A] [answered]
Hello, 

I am trying aRtsy and want to create a diamonds canvas that uses the color of the lowest frequent color of the data frame as a background. Is there any way to do this?

Currently, I have the following: I create a palette of 15 colors and then I select a random number between 1 and 6, this number is used to select how many colors I will take from the palette (also randomly). Right now I have a black background but would like to process the database so that I can use the lowest appearing color as a background.
```
set.seed(1)
palette2 <- distinctColorPalette(k = 15, altCol = FALSE, runTsne = FALSE)
cols = round(runif(1, 3, 6),0)
pal = palette2[round(runif(cols,1,15),0)]
back = 'black'

canvas_diamonds(colors = pal, 
            background = back, p = 0, col.line = back,
             resolution = 1000)
```
I was able to extract the color with the following code, but as I am using it after generating the plot, if I select this color and re run the code, it no longer matches the original. And I can't find a way to do it directly inside the canvas_diamonds function
```
data.frame(col = as.factor(p$data$col)) %>% 
  group_by(col) %>%
  summarise(n = n()) %>%
  slice(which.min(n)) %>%
  select(col)
```
I am trying this

```
canvas_diamonds(colors = pal, 
            background = data.frame(col = as.factor(.$data$col)) %>% 
              group_by(col) %>%
              summarise(n = n()) %>%
              slice(which.min(n)) %>%
              select(col), p = 0, col.line = back,
             resolution = 1000)
```

**Comment by koenderks** [comment, marked as answer]
Hi,

Cool that you are trying this out! It's probably best to first make the artwork with `NA` as the background color and `NA` as the line color. After that I would extract which color of the diamonds is the least frequent and set the background color with `theme_canvas()` (this is also applied internally). 

An example is given below:

```
set.seed(1)
palette2 <- randomcoloR::distinctColorPalette(k = 15, altCol = FALSE, runTsne = FALSE)
cols = round(runif(1, 3, 6),0)
pal = palette2[round(runif(cols,1,15),0)]

p <- aRtsy::canvas_diamonds(colors = pal, 
                     background = NA, p = 0, col.line = NA,
                     resolution = 1000)

# Extract least frequent color
back <- names(which.min(table(p$data$col)))

# Set this color as the background
p <- aRtsy::theme_canvas(x = p, background = back)
p
```

Hope this works for you!

**Reply by unbrother** [reply]
Thank you so much, this was exactly what I was looking for.
